File: src/test/resources/test_languages/js/code.js

// VIASH START
let par = {
  'input': 'input.txt',
  'real_number': 123.987654,
  'whole_number': 17,
  's': '...',
  'truth': true,
  'output': 'output.txt',
  'log': 'log.txt',
  'optional': 'help',
  'optional_with_default': 'me'
}
let meta = {
	'resources_dir': '.'
}
// VIASH END


// define helper functions
const fs = require('fs')

let logFun;
if (!par['log']) {
	logFun = function(out) {
	  console.log("INFO:" + out);
	}
} else {
	logFun = function(out) {
		fs.appendFileSync(par['log'], "INFO:" + out + "\n");
	}
}
let outFun;
if (!par['output']) {
	outFun = console.log
} else {
	outFun = function(out) {
		fs.appendFileSync(par['output'], out + "\n");
	}
}

// process parameters
logFun('Parsed input arguments.')

if (!par['output']) { 
	logFun('Printing output to console')
} else {
	logFun('Writing output to file')
}

for (const key in par) {
	if (Array.isArray(par[key]) && par[key].length == 0)
		outFun(`${key}: ||`)
	else if (Array.isArray(par[key]))
		outFun(`${key}: |${par[key].join(';')}|`)
	else if (par[key] == undefined)
		outFun(`${key}: ||`)
	else
		outFun(`${key}: |${par[key]}|`)
}

let inputData = fs.readFileSync(par['input'], 'utf8');
outFun(`head of input: |${inputData.split('\n')[0]}|`)
let resourceData = fs.readFileSync(meta['resources_dir'] + '/resource1.txt', 'utf8');
outFun(`head of resource1: |${resourceData.split('\n')[0]}|`)

for (const key in meta) {
	if (meta[key] == undefined || String(meta[key]) == 'NaN')
		outFun(`meta_${key}: ||`)
	else
		outFun(`meta_${key}: |${meta[key]}|`)
}

File: src/test/resources/test_languages/python/code.py

### VIASH START
par = {
  'input': 'input.txt',
  'real_number': 123.987654,
  'whole_number': 17,
  's': '...',
  'truth': True,
  'output': 'output.txt',
  'log': 'log.txt',
  'optional': 'help',
  'optional_with_default': 'me'
}
meta = {
  'resources_dir': '.'
}
### VIASH END

import logging
import sys

if par['log'] is not None:
	logging.basicConfig(filename=par['log'],level=logging.INFO)
else:
	logging.basicConfig(stream=sys.stdout,level=logging.INFO)

logging.info('Parsed input arguments.')

if par['output'] is not None:
    logging.info('Writing output to file')
    output_file = open(par['output'], 'w')
else:
    logging.info('Printing output to console')

def echo(s):
    if par['output'] is not None:
        output_file.write(s + "\n")
    else:
        print(s)

try:
    for key, value in par.items():
        if isinstance(value, list):
            if not value:
                echo(f"{key}: |empty array|")
            else:
                if isinstance(value[0] , bool):
                    value2 = map(lambda v: str(v).lower(), value)
                else:
                    value2 = map(lambda v: str(v), value)
                echo(f"{key}: |{';'.join(value2)}|")
        elif value is None:
            echo(f"{key}: ||")
        elif isinstance(value, bool):
            echo(f"{key}: |{str(value).lower()}|")
        else:
            echo(f"{key}: |{value}|")

    for key, value in meta.items():
        echo(f"meta_{key}: |{value}|")

    with open(par['input'], 'r') as infile:
        input = infile.readline().strip()
        echo(f"head of input: |{input}|")
    with open(meta['resources_dir'] + '/resource1.txt', 'r') as infile:
        resource = infile.readline().strip()
        echo(f"head of resource1: |{resource}|")
        
finally:
    if par['output'] is not None:
        output_file.close()